webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoComplete.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon, OverlayTrigger, Tooltip, TypeAheadSelect } from 'patternfly-react';
+import { STATUS } from '../../constants';
+import { bindMethods, debounceMethods } from '../../common/helpers';
+import './auto-complete.scss';
+
+class AutoComplete extends React.Component {
+  constructor(props) {
+    super(props);
+    bindMethods(this, ['handleClear', 'handleInputChange', 'handleItemSelect', 'handleInputFocus', 'getResults']);
+    debounceMethods(this, 250, ['handleInputChange', 'handleInputFocus', 'handleLoading']);
+  }
+
+  getResults(query, trigger) {
+    const { getResults, data: { controller, url } } = this.props;
+    getResults(url, query, controller, trigger);
+  }
+
+  handleInputFocus({ target: { value } }) {
+    // Only on initial focus.
+    if (this.props.results.length === 0) {
+      this.getResults(value, 'Input focus');
+    }
+  }
+
+  handleInputChange(query) {
+    this.getResults(query, 'Input changed');
+  }
+
+  handleItemSelect(selectedOptions) {
+    this.getResults(selectedOptions[0], 'Item selected');
+    /**
+       *  HACK: I had no choice but to call to an inner function,
+       * due to lack of design in react-bootstrap-typeahead.
+        */
+    this._typeahead.getInstance()._showMenu();
+  }
+
+  handleClear() {
+    this._typeahead.getInstance().clear();
+    this.getResults('', 'Input cleared');
+  }
+
+  handleLoading() {
+    return this.props.status === STATUS.PENDING;
+  }
+
+  componentWillUnmount() {
+    this.props.resetData();
+  }
+
+  render() {
+    const {
+      data,
+      results,
+      searchQuery,
+      menu,
+    } = this.props;
+
+    const tooltip = (<Tooltip id="__clear-tooltip__">{__('Clear')}</Tooltip>);
+    return (
+      <div>
+        <TypeAheadSelect
+          ref={(ref) => { this._typeahead = ref; }}
```

Comment:
ref API has been changed in react 16.3
https://reactjs.org/blog/2018/03/29/react-v-16-3.html